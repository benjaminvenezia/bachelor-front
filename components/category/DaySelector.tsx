import { View, Text, Pressable, StyleSheet } from "react-native";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { addDay, removeDay } from "../../store/slices/daysToAddTasksSlice";
import { Day } from "../../types/Day";
import { RootState } from "../../store/store";
import { GlobalStyles } from "../../constants/style";

type Props = {
  label: string | any;
};

const DaySelector = ({ label }: Props) => {
  const dispatch = useDispatch();
  const [isSelected, setIsSelected] = useState(false);

  const daysSelected = useSelector((state: RootState) => state.daysToAddTasks);

  const addTaskDay = () => {
    setIsSelected(!isSelected);

    if (isSelected) {
      dispatch(removeDay({ label: label }));
    }

    if (!daysSelected["activeDays"].includes(label)) {
      dispatch(addDay({ label: label }));
    }
  };

  return (
    <Pressable onPress={() => addTaskDay()} style={[styles.container, isSelected ? styles.isSelectedItem : null]}>
      <View>
        <Text style={[isSelected ? styles.isSelectedText : styles.text]}>{label}</Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 40,
    paddingVertical: 15,
    borderWidth: 5,
    borderColor: "white",
    marginVertical: 5,
    marginHorizontal: 3,
    borderRadius: 100,
  },
  isSelectedItem: {
    backgroundColor: GlobalStyles.colors.primary,
  },
  isSelectedText: {
    color: "black",
    fontSize: GlobalStyles.fontsSize.text,
  },
  text: {
    color: GlobalStyles.colors.text,
    fontSize: GlobalStyles.fontsSize.text,
  },
});

export default DaySelector;
