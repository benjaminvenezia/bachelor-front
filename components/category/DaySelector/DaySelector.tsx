import { View, Text, Pressable, StyleSheet } from "react-native";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { addDay, removeDay } from "../../../store/slices/daysToAddTasksSlice";
import { RootState } from "../../../store/store";
import { GlobalStyles } from "../../../constants/style";
import { DaySelectorProps } from "./DaySelectorProps.types";

const DaySelector = ({ label }: DaySelectorProps) => {
  const dispatch = useDispatch();
  const [isSelected, setIsSelected] = useState(false);

  const { activeDays } = useSelector((state: RootState) => state.daysToAddTasks);

  const addTaskDay = () => {
    setIsSelected(!isSelected);

    if (isSelected) {
      dispatch(removeDay({ label: label }));
    }

    if (!activeDays.includes(label)) {
      dispatch(addDay({ label: label }));
    }
  };

  return (
    <Pressable onPress={() => addTaskDay()} style={[styles.container, isSelected ? styles.isSelectedItem : null]}>
      <View>
        <Text style={[isSelected ? styles.isSelectedText : styles.text]}>{label}</Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 40,
    paddingVertical: 15,
    borderWidth: 5,
    borderColor: "white",
    marginVertical: 5,
    marginHorizontal: 3,
    borderRadius: 100,
  },
  isSelectedItem: {
    backgroundColor: GlobalStyles.colors.primary,
  },
  isSelectedText: {
    color: "black",
    fontSize: GlobalStyles.fontsSize.text,
  },
  text: {
    color: GlobalStyles.colors.text,
    fontSize: GlobalStyles.fontsSize.text,
  },
});

export default DaySelector;
