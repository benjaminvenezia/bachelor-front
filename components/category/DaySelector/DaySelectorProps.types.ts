export type DaySelectorProps = {
  label: string;
};
