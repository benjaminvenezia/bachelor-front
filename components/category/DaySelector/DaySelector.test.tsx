import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import DaySelector from "./DaySelector";
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe("<DaySelector />", () => {
    let store : any;
  
  beforeEach(() => {
    store = mockStore({ daysToAddTasks: { activeDays: [] } });
  });
  
  test("renders correctly", () => {
    const tree = renderer.create(<Provider store={store}><DaySelector label="Lun" /> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});