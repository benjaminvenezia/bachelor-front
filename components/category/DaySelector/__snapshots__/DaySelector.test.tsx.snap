// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaySelector /> renders correctly 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderColor": "white",
          "borderRadius": 100,
          "borderWidth": 5,
          "marginHorizontal": 3,
          "marginVertical": 5,
          "paddingHorizontal": 40,
          "paddingVertical": 15,
        },
        null,
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 20,
            },
          ]
        }
      >
        Lun
      </Text>
    </View>
  </View>,
  " ",
]
`;
